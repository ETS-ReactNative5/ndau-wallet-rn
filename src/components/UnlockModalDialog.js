import React, { Component } from 'react'

import { StyleSheet, Text, View } from 'react-native'
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen'
import ModalDialog from './ModalDialog'
import CommonButton from '../components/CommonButton'
import cssStyles from '../css/styles'

class UnlockModalDialog extends Component {
  _unlock = () => {
    this.closeModal()
  }

  showModal = () => {
    this._modalDialog.showModal()
  }

  closeModal = () => {
    this._modalDialog.closeModal()
  }

  render () {
    return (
      <ModalDialog
        ref={component => (this._modalDialog = component)}
        {...this.props}
      >
        <Text style={styles.text}>
          Unlocking this account means you will be able to spend from it, but it
          will no longer accrue an incentive (EAI). Are you sure you want to
          unlock?
        </Text>
        <Text style={styles.text}>
          Funds from this account till be available to you in 90 days. Until 90
          days are up you will not be able to add new ndau to this account.
        </Text>
        <View style={cssStyles.footer}>
          <CommonButton onPress={this._unlock} title='Start unlock countdown' />
        </View>
      </ModalDialog>
    )
  }
}

var styles = StyleSheet.create({
  text: {
    color: '#ffffff',
    fontSize: 20,
    fontFamily: 'TitilliumWeb-Regular',
    marginTop: hp('1%'),
    marginBottom: hp('1%'),
    marginLeft: wp('1%'),
    marginRight: wp('1%')
  }
})

export default UnlockModalDialog
