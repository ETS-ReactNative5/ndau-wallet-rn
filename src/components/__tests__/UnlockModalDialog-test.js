import { mount } from 'enzyme';
import React from 'react';
import { StyleSheet } from 'react-native';
import UnlockModalDialog from '../UnlockModalDialog';

import renderer from 'react-test-renderer';

describe('testing UnlockModalDialog...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  });
  const navigator = {
    setStyle: () => {}
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(<UnlockModalDialog parentStyles={styles} navigator={navigator} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
