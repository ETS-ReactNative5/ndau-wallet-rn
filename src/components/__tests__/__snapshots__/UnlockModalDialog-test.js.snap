// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UnlockModalDialog... renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  navigator={
    Object {
      "setStyle": [Function],
    }
  }
  onRequestClose={[Function]}
  parentStyles={
    Object {
      "wizardText": Object {
        "color": "#ffffff",
        "fontSize": 20,
      },
    }
  }
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#333333",
            "borderRadius": 3,
            "height": 934,
            "paddingBottom": 26.5,
            "paddingLeft": 15,
            "paddingRight": 15,
            "width": 675,
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#A9A9A9",
                "fontSize": 20,
              },
              Object {
                "marginLeft": 7.5,
                "marginTop": 6.5,
              },
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          ÔÄç
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "TitilliumWeb-Regular",
            "fontSize": 20,
            "marginBottom": 13.5,
            "marginLeft": 7.5,
            "marginRight": 7.5,
            "marginTop": 13.5,
          }
        }
      >
        Unlocking the account means you will be able to spend from it, but it will no longer accrue on incentive (EAI). Are you sure you want to unlock?
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "TitilliumWeb-Regular",
            "fontSize": 20,
            "marginBottom": 13.5,
            "marginLeft": 7.5,
            "marginRight": 7.5,
            "marginTop": 13.5,
          }
        }
      >
        Funds from this account till be available to you in 90 days. Until 90 days are up you will not be able to add new ndau to this account.
      </Text>
      <View
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityTraits="button"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "#4e957a",
              "borderRadius": 3,
              "height": 93.5,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                null,
                Object {
                  "color": "#fff",
                  "fontFamily": "TitilliumWeb-Regular",
                  "fontSize": 16,
                },
                null,
              ]
            }
          >
            Start unlock countdown
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
